compiler/src/iree/compiler/DispatchCreation/FormDispatchRegions.cpp

```diff
@@ -763,15 +775,20 @@ decideFusableLinalgOps(Region &region, DominanceInfo const &dominanceInfo,
       // If it is part of a fusion group or root op, ignore it.
       if (hasFusionGroupsAttribute(&op) || hasRootOpAttribute(&op))
         continue;
+      // If the operation is used for masking an AttentionOp, then we always
+      // clone it. The Attention mask is usually big, and is always generated
+      // from a small tensor, so it's always good to clone it.
+      if (isAttentionMaskGenerator(&op)) {
+        continue;
+      }
       // Only look for Linalg ops here. Avoid moving `linalg.fill` that aren't
       // fused with anything else into their own dispatches since it is better
       // to convert them to splats. Also avoid moving dequantization-like ops
       // into their own dispatch since it is better to clone these ops and avoid
       // materializing large tensors between dispatches.
       if (!isa<linalg::LinalgOp, tensor::PadOp, tensor::PackOp,
                IREE::Encoding::SetEncodingOp>(op) ||
-          isa<linalg::FillOp>(op) || IREE::LinalgExt::isBitExtendOp(&op) ||
-          IREE::LinalgExt::isGatherlikeOp(&op)) {
+          IREE::Flow::isClonableIntoDispatchOp(&op)) {
```

Comment:
Interesting. If we are going to do this, then won't it be easier to just make that method check if the op is attention mask and not have to check it everywhere?

Reply:
I think that's what we are already doing right? The attention mask checking logic is in IREE::Flow::isClonableIntoDispatchOp.

Reply:
Actually, i got what you meant. I removed the addition mask check here and only rely on the isClonableIntoDispatch check now.